File: leetcode/rust-leetcode/src/2022.convert-1d-array-into-2d-array.rs
impl Solution {
    pub fn construct2_d_array(original: Vec<i32>, m: i32, n: i32) -> Vec<Vec<i32>> {
        let (m, n) = (m as usize, n as usize);
        if original.len() != m*n {
            vec![]
        } else {
            let mut ans = vec![];
            for i in 0..m {
                ans.push(original[i*n..(i+1)*n].to_vec());
            }
            ans
        }
    }
}

File: leetcode/python/2022.convert-1d-array-into-2d-array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if len(original) != m * n:
            return []
        ans = [[0] * n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                ans[i][j] = original[i * n + j]
        return ans

